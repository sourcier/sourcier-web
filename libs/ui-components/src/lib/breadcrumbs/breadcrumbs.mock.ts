export const path = '/home/documents/add-document';
