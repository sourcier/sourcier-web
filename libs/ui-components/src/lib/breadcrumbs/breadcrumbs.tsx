import { VscFolder, VscFile } from 'react-icons/vsc';

export interface BreadcrumbsProps {
  path: string;
  leafRewrite?: { pattern: string; replacement: string };
  leafAsFolder?: boolean;
}

export function Breadcrumbs({ path, leafAsFolder }: BreadcrumbsProps) {
  const crumbs = path.split('/').filter(Boolean);
  const labels = crumbs.map((crumb) => crumb.replace(/-/g, ' '));
  const links = crumbs.map((crumb, index, array) =>
    array.slice(0, index + 1).join('/')
  );

  return (
    <div className="text-lg breadcrumbs">
      <ul>
        {crumbs.map((crumb, index, array) => {
          return (
            <li key={index}>
              <a className="capitalize" href={`/${links[index]}`}>
                {index === array.length - 1 && !leafAsFolder ? (
                  <VscFile className="mr-2" />
                ) : (
                  <VscFolder className="mr-2" />
                )}
                {labels[index]}
              </a>
            </li>
          );
        })}
      </ul>
    </div>
  );
}

export default Breadcrumbs;
