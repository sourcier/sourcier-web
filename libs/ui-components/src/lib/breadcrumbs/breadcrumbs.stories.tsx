import { Story, Meta } from '@storybook/react';
import { Breadcrumbs, BreadcrumbsProps } from './breadcrumbs';

import { path } from './breadcrumbs.mock';

export default {
  component: Breadcrumbs,
  title: 'Breadcrumbs',
} as Meta;

const Template: Story<BreadcrumbsProps> = (args) => <Breadcrumbs {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  path,
};
