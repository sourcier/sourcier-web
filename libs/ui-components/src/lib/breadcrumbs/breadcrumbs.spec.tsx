import renderer from 'react-test-renderer';

import Breadcrumbs from './breadcrumbs';
import { path } from './breadcrumbs.mock';

describe('Breadcrumbs', () => {
  it('should render successfully', () => {
    const tree = renderer.create(<Breadcrumbs path={path} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
