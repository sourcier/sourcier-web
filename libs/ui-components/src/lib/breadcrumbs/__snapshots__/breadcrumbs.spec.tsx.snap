// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render successfully 1`] = `
<div
  className="text-lg breadcrumbs"
>
  <ul>
    <li>
      <a
        className="capitalize"
        href="/home"
      >
        <svg
          className="mr-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.5 3H7.71l-.85-.85L6.51 2h-5l-.5.5v11l.5.5h13l.5-.5v-10L14.5 3zm-.51 8.49V13h-12V7h4.49l.35-.15.86-.86H14v1.5l-.01 4zm0-6.49h-6.5l-.35.15-.86.86H2v-3h4.29l.85.85.36.15H14l-.01.99z"
          />
        </svg>
        home
      </a>
    </li>
    <li>
      <a
        className="capitalize"
        href="/home/documents"
      >
        <svg
          className="mr-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.5 3H7.71l-.85-.85L6.51 2h-5l-.5.5v11l.5.5h13l.5-.5v-10L14.5 3zm-.51 8.49V13h-12V7h4.49l.35-.15.86-.86H14v1.5l-.01 4zm0-6.49h-6.5l-.35.15-.86.86H2v-3h4.29l.85.85.36.15H14l-.01.99z"
          />
        </svg>
        documents
      </a>
    </li>
    <li>
      <a
        className="capitalize"
        href="/home/documents/add-document"
      >
        <svg
          className="mr-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M13.71 4.29l-3-3L10 1H4L3 2v12l1 1h9l1-1V5l-.29-.71zM13 14H4V2h5v4h4v8zm-3-9V2l3 3h-3z"
            fillRule="evenodd"
          />
        </svg>
        add document
      </a>
    </li>
  </ul>
</div>
`;
